dts/bindings/wifi/wifi-tx-power-5g.yaml

```diff
@@ -0,0 +1,37 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  This is a representation of the nRF70 series transmit power ceilings
+  i.e., maximum transmit power allowed for 5GHz band.
+
+  low sub-band: Channels 36-64
+  mid sub-band: Channels 96-132
+  high sub-band: Channels 136-177
+
+  Resolution is 0.25 dBm
```

Comment:
We don't need that level of granularity for Wi-Fi TX power, 0.25 is good enough to cover all cases.

Reply:
We may not, but someone else might, this is a generic binding, so we should be "overly" granular here so it is useful to everyone :) One can always round to nearest, not the opposite :)

Reply:
When I say `we` I was referring to all Wi-Fi chips, in general Wi-Fi chips don't need that granularity.

>  so we should be "overly" granular here so it is useful to everyone :)

This is the problem with making things general but with only a single user :), we end up over engineering without a second user for a long time and even if we had second user we might end up with a diverge set of requirements needing a redesign making the effort to make generic changes moot (YAGNI) :).

Reply:
IMHO, as this is not defined by Wi-Fi standard, whatever value you put in there would be `odd`, for me `0.1/0.0001/0.25` are all odd :).

https://openwrt.org/docs/guide-user/network/wifi/basic#common_options which is relevant to Wi-Fi sets it to `1dBM` as the unit.

So, if you ask me I would just make it `dBM` as the units, but please note that this is a breaking change as all existing customers need to migrate to this scale.

Reply:
In fact these are not absolute power levels, these are maximum TX power limits, so, we don't need granularity at all as we can round them off.

Reply:
I'm okay with whole `dBm` for simplicity, given the argument that granularity is not required for tx power limits.

Reply:
> but please note that this is a breaking change as all existing customers need to migrate to this scale. So, I would prefer to leave this untouched.

What about this? In upstream this is the initial support, so, it doesn't break anything, but as customers are already using it using the NCS, this would break the expectation, can we please table this at a later time till we figure out how to update seamlessly?

Reply:
These options are in the devicetree, which we are already changing massively, downstream users will have to adapt. Better make all breaking changes now in one go rather than forcing breaking changes later.

Reply:
Sure, but others are obvious to find i.e., cause build errors, but this would be a silent change affecting functionality and would be tough to root cause. Anyways, for a quicker resolution of this thread, let me change to `dBm` and handle this in release notes and hope for the best.